orion-server/src/test/java/com/pinterest/orion/core/automation/sensor/kafka/KafkaClusterInfoSensorTest.java

```diff
@@ -0,0 +1,99 @@
+package com.pinterest.orion.core.automation.sensor.kafka;
+
+import com.google.gson.stream.MalformedJsonException;
+import com.pinterest.orion.core.kafka.Brokerset;
+import com.pinterest.orion.core.kafka.Brokerset.BrokersetRange;
+import com.pinterest.orion.core.kafka.BrokersetTest;
+import org.assertj.core.api.Assertions;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+public class KafkaClusterInfoSensorTest {
```

Comment:
Is there a test that covers the end to end broker replacement scenario? I.e.
1. There is a given brokerset that is affected by a broker-range replacement
2. Override brokerset is created
3. Replacement is complete
4. Override brokerset is removed
5. Before and after brokersets (ranges) are the same.

Reply:
Hmm, not a bad idea, but I don't know how useful it would be here, as each of those operations are idempotent. I mean the way I can think to set up a test like that here would be to have 
`List<Brokerset> result = KafkaClusterInfoSensor.loadBrokersetsFromFile(brokersetFile, overridesFile);`
then 
`List<Brokerset> result2 = KafkaClusterInfoSensor.loadBrokersetsFromFile(brokersetFile);`
then 
`List<Brokerset> result3 = KafkaClusterInfoSensor.loadBrokersetsFromFile(brokersetFile, overridesFile);`
Which the test is going to look at and be perfectly happy to tell me that the two identical calls to an idempotent method are equal. The only thing that would really be testing would be the file system operations at that point (Which if I was going to make these tests better would be the first things I would remove)
That test would be better for a big scary integration test, which at this point I am not prepared to get into.

Reply:
And as for getting rid of the tests, if I was told to do that I would have to MIGHTILY resist the urge to convert this test to Kotlin, because that would be another conversation that I don't want to have right now.